From 56ccb67a33acca1cd6b8007076399c347964bf74 Mon Sep 17 00:00:00 2001
From: Corey Berla <corey@berla.me>
Date: Tue, 27 Sep 2022 16:04:30 -0700
Subject: [PATCH] list-base: Return correct order in real_get_selection()

We are using g_list_prepend() in real_get_selection(), so it should
be reversed.

Fixes: https://gitlab.gnome.org/GNOME/nautilus/-/issues/2512
---
 src/nautilus-list-base.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/nautilus-list-base.c b/src/nautilus-list-base.c
index 3e4bd234f..86dbd6db2 100644
--- a/src/nautilus-list-base.c
+++ b/src/nautilus-list-base.c
@@ -1135,6 +1135,8 @@ real_get_selection (NautilusFilesView *files_view)
                                          g_object_ref (nautilus_view_item_get_file (item)));
     }
 
+    selected_files = g_list_reverse (selected_files);
+
     return selected_files;
 }
 
-- 
GitLab

